### ISSUE TITLE
Add "Shared With" column to Past Goals list

### ISSUE THREAD

**Issue by Travis-A-Clark** [bcgov/performance#960]
**Current Situation**

- There is a a Shared With column on the Current Goals page, but not on the Past Goals page. 
- When a goal is shared with another employee, that goal shows up in their "Goals Shared with Me" tab when it is active but then disappears when it is achieved or archived.

**Desired Solution**

- The columns for Current and Past Goals lists should be the same.
- When a shared goal status is changed to achieved or archived, it should move to the Past goals list for all employees who had access to it.

**Comment by steffipravasi** [comment]
Any shared goal when marked Achieved or Archived is moved to My Past Goals for the user who has created the goal, but not for the employees with whom the goal is shared. 
A goal named BakuskoMarkGoal is shared with Chad Hoskins  and Barbara Briault. Chad and Barbara cannot see the goal in My Past Goals list
Marks' profile:![Screenshot 2023-07-07 at 2.19.16 AM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/fc76c884-5397-46b7-9665-11bd8adafabd)
Barbaras' profile:![Screenshot 2023-07-07 at 2.21.44 AM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/f5c4a971-dfc0-4959-bf0c-477376b398e9)

@Travis-A-Clark If we'll display the goal in the past goals list for all users who had access to it then the shared employee, can mark the goal as active and the goal will be displayed in My current goals page regardless if the goal was copied from the Goals Shared With Me page or not. We might need a little more clarification here.

**Comment by jizhaogit** [comment]
based on what i found in the change history,  the shared goal was not listed neither in "My Current Goal" nor "My Shared Goal". these two list are just for the goal owner him/her self items. All the **active** goals that be shared to the user are listed in the "Goals Shared with Me".
@Travis-A-Clark please help clarify if we are going to change this. thanks!

**Comment by Travis-A-Clark** [comment]
@steffipravasi @jizhaogit 

Sorry for the confusion!

The desired outcome would be:
1) Travis creates a goal; it shows up in his Current Goals list
2) Travis shares that goal with Cathy; it shows up in Cathy's "Goals Shared with Me" list
3) Travis can edit the content of the goal and change its status (achieved, archived); Cathy can only view the goal and add comments.
4) Let's remove the option for Cathy to "copy" the goal. She can only view and add comments.
5) When Travis changes the goal status to "achieved" or "archived", the goal moves to his Past Goals list; it also moves to Cathy's Past Goals list.
6) Ideally, Cathy would only be able to view this goal, not change the status back to active.

Basically, Travis is the goal owner and he can edit and change status. He shares the goal with Cathy which grants her view and comment access.

Is that possible? Thanks!

**Comment by jizhaogit** [comment]
a new fix pushed in test for the Travis clarifications

**Comment by steffipravasi** [comment]
Goal owned by Mark ![Screenshot 2023-07-11 at 4.18.05 PM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/7d028c61-5f1f-4cc3-aedb-5fcc9481a50c)
Goal displayed for Shawna and Dave![Screenshot 2023-07-11 at 4.19.03 PM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/504d1319-2ae3-4f72-a5b0-a18a93a04994)
- What should the 'Shared with' section for Shawna display? 
- Should they have an Delete goal option?

**Comment by steffipravasi** [comment]
@jizhaogit Assigning ticket to you to remove the Delete button

**Comment by steffipravasi** [comment]
Delete button has been removed from the Past Goals for users with whom the goal is shared.![Screenshot 2023-07-11 at 5.39.06 PM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/1f2ce248-8b29-4816-902f-80861378d8bb)
Also, only comment to the goal is allowed to the shared users. 
Ready for PO review.
